index.bs

```diff
@@ -113,186 +113,206 @@ div.enum-description > table > thead > tr > th {
       Note: The autoplay policy represents the current status of whether a
       user-agent allows media to autoplay, which can **vary** in the future.
       Therefore, it is **recommended** that authors check the result every time
-      they want to have an up-to-date result.
+      if they want to have an up-to-date result.
     </div>
 
     <div class=example>
       If a user-agent uses the user activation, described in
       [[HTML#user-activation-data-model]], to determine if the autoplay media
-      are allowed or not, and it blocks all autoplay by default
-      ({{AutoplayPolicy/disallowed}}). Then the policy can change to
+      are allowed or not, and the default policy is to blocks all autoplay
+      ({{AutoplayPolicy/disallowed}}). Then the policy could change to
       {{AutoplayPolicy/allowed}} or {{AutoplayPolicy/allowed-muted}} after a
       user performs a supported user gesture on the page or the media.
     </div>
 
-  <h3 id="document-api">The Document API</h3>
-    <pre class="idl">
-    partial interface Document {
-      readonly attribute AutoplayPolicy autoplayPolicy;
-    };
-    </pre>
-    This represents a rough status of whether media that belong to this
-    {{Document}} are allowed to autoplay or not.
+  <h3 id="autoplay-detection-methods">The Autoplay Detection Methods</h3>
+    <h4 id="query-by-a-constructor">Query by a Constructor</h4>
+      <pre class="idl">
+        callback HTMLMediaElementConstructor = HTMLMediaElement ();
+        callback AudioContextConstructor = AudioContext ();
+
+        [Exposed=Window]
+        partial interface Navigator {
+          AutoplayPolicy getAutoplayPolicy(HTMLMediaElementConstructor ctor);
+
+          AutoplayPolicy getAutoplayPolicy(AudioContextConstructor ctor);
```

Comment:
One problem I would like to discuss here is, if a web developer passes a contructor which is not `HTMLMediaElementConstructor` and `AudioContextConstructor`, should we throw an exception? 

Same problem for query-by-a-element case as well, if a web developer passes other elements which we don't support.